### PR TITLE
Sprint 2

### PR DESCRIPTION
**Pull request by Oupsla**
Intégration d'une liste de beacons grâce à Proximy
Ajout de position aux produits
Calcul d'une trilateration + moyenne de la position

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
 node_modules
 coverage
-
+dist
```
Comment (duribreux) [inline .eslintignore:3]: Also add the dist folder to the .gitignore file and explain steps required to generate it.

File: model/beacon/beacon-facade.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const Model = require('../../lib/facade');
+const beaconSchema  = require('./beacon-schema');
+
+class BeaconModel extends Model {
+
+  updateBeacons(beaconsList) {
+    return beaconSchema
+      .remove({})
```
Comment (duribreux) [inline model/beacon/beacon-facade.js:8]: Remove `{}` ?

File: model/product/product-facade.js [hunk 3]
```diff
@@ -10,6 +12,67 @@ class ProductModel extends Model {
     .remove()
     .exec();
   }
+
+  addPosition(eanQuery, positions) {
```
Comment (duribreux) [inline model/product/product-facade.js:16]: Split a bit this methods, that's a lot of things done in it

File: util/trilateration.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+
+// Requires the Mathjs library - http://mathjs.org/
+const math = require('mathjs');
+
+/**
+ * Represents a coordinate with a distance
+ * @param {Number} lat  Latitude
+ * @param {Number} lon  Longitude
+ * @param {Number} dist Distance from [lat, lon] to some point in kilometers
```
Comment (duribreux) [inline util/trilateration.js:12]: I don't really understand the semantic behind the distance in the beacon.
(x,y) coordinates are easy to figure out but distance... "from some point in km" ? Makes no sense to me. What point ? Is this really the best place to save this value ?

File: util/trilateration.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+};
+
+/**
+ * Perform a trilateration calculation to determine a location
+ * based on 3 beacons and their respective distances (in kilometers) to the desired point.
+ *
+ * @param  {Array} beacons Array of 3 Beacon objects
+ * @return {Array}         Array of the format [latitude, longitude]
+ */
+const trilaterate = function(beacons) {
```
Comment (duribreux) [inline util/trilateration.js:29]: No libs to that ? oO'

**Review by duribreux — CHANGES_REQUESTED** [review]

File: model/beacon/beacon-facade.js (hunk 2 above)
Comment (Oupsla) [reply model/beacon/beacon-facade.js:8]: This method has been created to update the entire list of beacon from the website proximy
So first I have to remove all beacons in db

File: .eslintignore (hunk 1 above)
Comment (Oupsla) [reply .eslintignore:3]: The dist folder is not generated, you only have to change the dist/swagger.yaml
You can use an editor like : http://editor.swagger.io/

File: util/trilateration.js (hunk 4 above)
Comment (Oupsla) [reply util/trilateration.js:12]: Thanks to proximy, we know where all beacons are (with their latitudes and longitudes).
The phone gives us a list of distance between detected beacons and the phone himself

So this method returns the approximative position of the user

File: util/trilateration.js (hunk 5 above)
Comment (Oupsla) [reply util/trilateration.js:29]: Only one but not trustable

**Pull request merged**